src/grafana_agent.py

```diff
@@ -288,10 +288,13 @@ def _requested_tracing_protocols(self) -> Set[ReceiverProtocol]:
         )
 
     def _update_tracing_provider(self):
+        requested_tracing_protocols = (
```

Comment:
I think a comment here explaining why we can't just publish right away will be useful to help us remember later

Reply:
True, added a comment :)